routes/api/profile.js

```diff
@@ -6,7 +6,50 @@ const { check, validationResult } = require("express-validator");
 const config = require("config");
 const auth = require("../../middleware/auth");
 
-router.get("/me", auth, (req, res) => {
-  res.json({test: "test"})
+// Get logged-in user's profile
+router.get("/me", auth, async (req, res) => {
+  try {
+    const profile = await Profile.findOne({user: req.user.id});
+    if(!profile) {
+      return res.status(404).json({msg: "Profile not found"})
+    }
+    res.json(profile)
+  } catch ({message}) {
+    response.status(500).send("There's a server error.")
+  }
+})
+
+// Create & update profile
+router.post("/", [auth, [check("location", "Location is required.").not().isEmpty()]], async (req, res) => {
+  const error = validationResult(req);
+    if (!error.isEmpty()) {
+      return res.status(400).json({
+        error: error.array(),
+      });
+    }
 
+  const {location} = req.body;
+  const updates = {
+    user: req.user.id,
+    location
+  }
```

Comment:
```suggestion
  const updates = {
    user: req.user.id,
    location,
    gender,
    bio,
    profession
    skills
  }
```
we need to update those values no matter whether users wanna change them.